db/compaction/compaction.cc

```diff
@@ -340,6 +344,13 @@ bool Compaction::IsTrivialMove() const {
     if (compaction_size > max_compaction_bytes_) {
       return false;
     }
+
+    if (partitioner.get() != nullptr) {
+      partitioner->Reset(file->smallest.user_key());
+      if (partitioner->ShouldPartition(file->largest.user_key())) {
```

Comment:
what's the use case for splitting SST when there's opportunity for trivial move? If you thing there's a valid case, can we add an option for it?

Reply:
Oh, you are right there can still be case where there is no need to split for some files. I will add some "CanDoTrivialMove(key_form, key_to)" function

Reply:
done